intranet/apps/auth/views.py

```diff
@@ -200,11 +200,13 @@ def post(self, request):
                     today_8 = Day.objects.today().day_type.blocks.filter(name__contains="8")
                     if today_8:
                         first_start_time = time(today_8[0].start.hour, today_8[0].start.minute)
+                        last_start_time = time(today_8.pop().start.hour, today_8.pop().start.minute)
```

Comment:
```suggestion
                        last_start_time = time(today_8[-1].start.hour, today_8[-1].start.minute)
```

`.pop()` would remove the element before returning it; causing the second `.pop()` to not return the same block as the first `.pop()`.